pkg/swarm/validator.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package swarm contains most basic and general Swarm concepts.
+package swarm
+
+import "context"
+
+type ValidatorFunc func(context.Context, Chunk) bool
+
+type Validator struct {
+	Valid    func(ch Chunk) (valid bool)
+	Callback func(context.Context, Chunk)
+}
+
+func NewValidator(vs ...Validator) ValidatorFunc {
```

Comment:
To be precise and clear, this function should be called `NewValidatorFunc` as it conflicts with the `Validator` interface here. I would even go and suggest these name changes:

`ValidatorFunc` -> `Func` (validator is the package name)
`NewValidator` -> `New`

or even 

`ValidatorFunc` -> interface type `Service`
`Validator` -> `NewService`

but that may be too much.